packages/cozy-client/src/helpers/urlHelper.js

```diff
@@ -201,18 +201,15 @@ const wellKnownUrl = url => uri(url) + '/.well-known/change-password'
  *   Cozy's root or to a specifc slug. The caller is responsible to handle that exception
  * - another status means there aren't any Cozy behind to the given origin
  *
- * @param {object} url          Object of URL elements
- * @param {string} url.protocol Protocol to use in the origin (e.g. http)
- * @param {string} url.hostname Hostname to use in the origin (e.g. claude.mycozy.cloud)
- * @param {string} url.port     Port to use in the origin (e.g. 8080)
+ * @param {URL} url URL to validate
  *
  * @returns {Promise<boolean>} True if we believe there's a Cozy behind the given origin
  * @throws {InvalidCozyUrlError} Thrown when we know for sure there aren't any Cozy behind the given origin
  * @throws {BlockedCozyError} Thrown when we know for sure there is Cozy behind the given origin but it is in a "Blocked" state
  */
 const isValidOrigin = async url => {
   const response = await fetch(wellKnownUrl(url))
-  const { status } = response
+  const { status, url: responseUri } = response
```

Comment:
I thought about using `redirect: 'error'` but a request to `https://example.mycozy.cloud` will redirect most of the time since the user will not be logged in at that point and we don't want to error out in this cas.

Reply:
Relying on `redirected` would not give us the expected behavior either since we don't want to error out on all redirections (see previous comment) but only those who change the `origin` value of the URL.